app/models/template_input.rb

```diff
@@ -2,11 +2,16 @@ class TemplateInput < ActiveRecord::Base
   class ValueNotReady < ::Foreman::Exception
   end
 
-  TYPES = { :user => N_('User input'), :fact => N_('Fact value'), :variable => N_('Variable'),
+  class ValueNotPermitted < ::Foreman::Exception
+  end
+
+  TYPES = { :user => N_('User input'), :select_options => N_('User selectable options'),
```

Comment:
It's not a "must" but I think it's very worthwhile. I made it just part of User Input now.